### PR TITLE
fix: long dropdown popup truncates items

### LINKED ISSUE #8008: [Bug] Property pane: long dropdown popup with many entries goes out of the viewport
## Description
Release/Production - when there are more number(25-30) of widgets in a page and setting up Onclick actions for any of the widgets it is observed that when the property pane is at the top of the canvas the widget selection pane goes beyond the screen an user is able to select widgets via keyboard and the hidden widgets are not selectable via mouse selection

Expected Result:
If there are more widgets then the during OnClick settings the widgets should not hide and should be easily selectable  

### Steps to reproduce the behaviour:

1. Launch Appmsith app in the release environment 
2. Drag and drop multipe widget (25-30)
3. Open and o the property pane and place the property pane at the top 
4. Set OnClick actions to reset Widget 
5. Choose the widget to reset and try to drag hidden widgets via mouse and observe

**Vdeolink:**
[LOOM DEMO](https://www.loom.com/share/7b4bb19092d14b1787730591f7321395)

### Important Details

- Version: [11.3]
- OS: [MacOS Big Sur]
- Browser [chrome, safari]
- Environment [production, release]

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

Set max height and overflow auto so that dropdown menu won't go out of viewport and content will be scrollable

Fixes #8008 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/dropdown-overflow 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | app/client/src/components/ads/TreeDropdown.tsx | 58.49 **(-1.13)** | 58.82 **(0)** | 36.84 **(-2.05)** | 56.86 **(-1.14)**</details>

### CODE REVIEW

**Review by somangshu — COMMENTED** [review]
I find the change alright. I have two concerns though:
1. With the docked property pane would this still be required @jsartisan 
2. This is an ADS component and may have a cascading effect. @marks0351 would you be able to confirm

**Comment by aswathkk** [comment]
@somangshu I also had the same concerns that you had.
1. I checked the DP for docked property pane and found that this is still required
2. I checked other places where `TreeDropdown` is used and found that this change is not affecting anywhere else

@marks0351 @jsartisan Let me know if I'm wrong.

**Comment by marks0351** [comment]
@aswathkk can u pls have a work with @Tooluloope he was working on pop up fixes #7499 , not sure if he has catered to this issue.

**Comment by Tooluloope** [comment]
> @aswathkk can u pls have a word with @Tooluloope he was working on pop up fixes #7499 , not sure if he has catered to this issue.

@marks0351 The other PR was for widget popup/dropdown, this PR is tailored towards the PropertyPane dropdown

File: app/client/src/components/ads/TreeDropdown.tsx [hunk 1]
```diff
@@ -44,6 +44,8 @@ type TreeDropdownProps = {
 };
 
 const StyledMenu = styled(Menu)`
+  max-height: calc(100vh - 64px);
```
Comment (Tooluloope) [inline app/client/src/components/ads/TreeDropdown.tsx:47]: @aswathkk whats the 64px for?
Could you leave a comment, and writing test would be ideal as well

**Review by Tooluloope — CHANGES_REQUESTED** [review]

File: app/client/src/components/ads/TreeDropdown.tsx (hunk 1 above)
Comment (aswathkk) [reply app/client/src/components/ads/TreeDropdown.tsx:47]: @Tooluloope It should've been 68px instead which is the total height of top bar and bottom bar.
I will update it and make it more readable.

File: app/client/src/components/ads/TreeDropdown.tsx [hunk 2]
```diff
@@ -44,6 +44,11 @@ type TreeDropdownProps = {
 };
 
 const StyledMenu = styled(Menu)`
+  max-height: calc(
+    100vh - ${(props) => props.theme.smallHeaderHeight} -
+      ${(props) => props.theme.bottomBarHeight}
+  );
```
Comment (Tooluloope) [inline app/client/src/components/ads/TreeDropdown.tsx:50]:
```suggestion
 max-height: ${(props) =>
    `calc( 100vh - ${props.theme.smallHeaderHeight} - ${props.theme.bottomBarHeight})`};
```

Comment (Tooluloope) [reply app/client/src/components/ads/TreeDropdown.tsx:50]: Might be better this way.

**Review by Tooluloope — DISMISSED** [review]

**Review by Tooluloope — APPROVED** [review]

**Comment by aswathkk** [comment]
I'm facing an issue with testing this in cypress.
This is related to a [resolved cypress issue](https://github.com/cypress-io/cypress/issues/4395) I guess.

@NandanAnantharamu Can you help me with this?

**Comment by NandanAnantharamu** [comment]
> resolved cypress issue

this is has been resolved now

**Comment by vivekverma2312** [comment]
@aswathkk Tested and working fine now.

**Review by Tooluloope — APPROVED** [review]

**Pull request merged**